pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -414,7 +421,34 @@ func isBestEffort(pod *corev1.Pod) bool {
 	return qos.GetPodQOS(pod) == corev1.PodQOSBestEffort
 }
 
+// IsTerminating returns true if the pod is actively being deleted.
+// Fixed in https://github.com/kubernetes/kubernetes/issues/135411
+//
+// Previously this incorrectly checked activeDeadlineSeconds, but that field
+// indicates a time limit, not termination status. The correct field to check
+// is deletionTimestamp, which is set when a pod deletion is triggered.
+//
+// This function is used by ResourceQuota's "Terminating" scope to identify
+// pods that are in the termination process (e.g., after kubectl delete).
 func IsTerminating(pod *corev1.Pod) bool {
+	// A pod is terminating when it has a deletion timestamp set.
+	// This indicates the pod is in the termination process (being deleted).
+	return pod.DeletionTimestamp != nil
```

Comment:
/hold

I don't think this is correct. non-nil spec.ActiveDeadlineSeconds OR non-nil metadata.DeletionTimestamp both limit the lifetime of the pod